instrumentation/opentelemetry-instrumentation-httpx/src/opentelemetry/instrumentation/httpx/__init__.py

```diff
@@ -0,0 +1,325 @@
+
+import typing
+
+import httpcore
+import httpx
+import wrapt
+
+from opentelemetry import context
+from opentelemetry.instrumentation.httpx.version import __version__
+from opentelemetry.instrumentation.instrumentor import BaseInstrumentor
+from opentelemetry.instrumentation.utils import (
+    http_status_to_status_code,
+    unwrap,
+)
+from opentelemetry.propagate import inject
+from opentelemetry.trace import SpanKind, Tracer, TracerProvider, get_tracer
+from opentelemetry.trace.span import Span
+from opentelemetry.trace.status import Status
+
+ResponseInfo = typing.Tuple[
+    int, typing.List[typing.Tuple[bytes, bytes]], typing.Iterable[bytes], dict,
+]
+NameCallback = typing.Callable[[str, str], str]
+SpanCallback = typing.Callable[[Span, ResponseInfo], None]
+URL = typing.Tuple[bytes, bytes, typing.Optional[int], bytes]
+Headers = typing.List[typing.Tuple[bytes, bytes]]
+
+
+def _get_tracer(
+    *, tracer_provider: typing.Optional[TracerProvider] = None
+) -> Tracer:
+    return get_tracer(
+        __name__,
+        instrumenting_library_version=__version__,
+        tracer_provider=tracer_provider,
+    )
+
+
+def _get_span_name(
+    method: str,
+    url: str,
+    *,
+    name_callback: typing.Optional[NameCallback] = None
+) -> str:
+    span_name = ""
+    if name_callback is not None:
+        span_name = name_callback(method, url)
+    if not span_name or not isinstance(span_name, str):
+        span_name = "HTTP {}".format(method).strip()
+    return span_name
+
+
+def _apply_status_code(span: Span, status_code: int) -> None:
+    if not span.is_recording():
+        return
+
+    span.set_attribute("http.status_code", status_code)
+    span.set_status(Status(http_status_to_status_code(status_code)))
+
+
+def _prepare_attributes(method: bytes, url: URL) -> typing.Dict[str, str]:
+    _method = method.decode().upper()
+    _url = str(httpx.URL(url))
+    span_attributes = {
+        "http.method": _method,
+        "http.url": _url,
+    }
+    return span_attributes
+
+
+def _prepare_headers(headers: typing.Optional[Headers]) -> httpx.Headers:
+    return httpx.Headers(headers)
+
+
+class SyncOpenTelemetryTransport(httpcore.SyncHTTPTransport):
+    """Sync transport class that will trace all requests made with a client.
+
+    Args:
+        transport: SyncHTTPTransport instance to wrap
+        tracer_provider: Tracer provider to use
+        span_callback: A callback provided with the response info to modify
+            the span
+        name_callback: A callback provided with the method and url to process
+            the span name
+    """
+
+    def __init__(
+        self,
+        transport: httpcore.SyncHTTPTransport,
+        tracer_provider: typing.Optional[TracerProvider] = None,
+        span_callback: typing.Optional[SpanCallback] = None,
+        name_callback: typing.Optional[NameCallback] = None,
+    ):
+        self._transport = transport
+        self._tracer_provider = tracer_provider
```

Comment:
could call tracer_provider and store tracer instead assuming all uses will have the instrumentation lib and version tags.

Reply:
Storing the tracer now.